### PR TITLE
feat: da issue 79

### PR DESCRIPTION
**Pull request by bushjames**
See https://github.com/mojaloop/design-authority/issues/79

In summary, if fspiop-destination header is present in a GET /parties/... request the ALS should respect it and forward the request to the specified DFSP without consulting any oracles.

### CODE REVIEW

File: test/unit/handlers/parties/{Type}/{ID}/{SubId}.test.js [hunk 1]
```diff
@@ -71,13 +71,17 @@ describe('/parties/{Type}/{ID}/{SubId}', () => {
     parties.getPartiesByTypeAndID.restore()
   })
 
-  it('getPartiesByTypeAndID endpoint sends async 3204 to /error for invalid party ID on response with status 400', async () => {
+  it.only('getPartiesByTypeAndID endpoint sends async 3204 to /error for invalid party ID on response with status 400', async () => {
```
Comment (lewisdaly) [inline test/unit/handlers/parties/{Type}/{ID}/{SubId}.test.js:74]: oops!

**Review by lewisdaly — CHANGES_REQUESTED** [review]

File: src/domain/parties/parties.js [hunk 2]
```diff
@@ -67,13 +67,31 @@ const getPartiesByTypeAndID = async (headers, params, method, query, span = unde
...
+        // first check the destination is a valid participant
+        const destParticipantModel = await participant.validateParticipant(headers[Enums.Http.Headers.FSPIOP.DESTINATION], childSpan)
+        if(!destParticipantModel) {
+          Logger.error('Destination FSP not found')
+          throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Destination FSP not found')
+        }
+
+        // all ok, go ahead and forward the request
+        await participant.sendRequest(headers, headers[Enums.Http.Headers.FSPIOP.DESTINATION], callbackEndpointType, Enums.Http.RestMethods.GET, undefined, options, childSpan);
+        return;
```
Comment (mdebarros) [inline src/domain/parties/parties.js:90]: I believe you should be closing out the `childSpan` and `histTimerEnd`.

I suggest that you should remove the `return` and rather add an "else" to your if-statement, and let those be handled by the "finally" of the try-catch.

Or alternatively:

```suggestion
        histTimerEnd({ success: true })
        if (childSpan && !childSpan.isFinished) {
          childSpan.finish()
        }
        return;
```

Additionally, you have a `semi-colon` at the end which I believe should fail the lint rules.

**Review by mdebarros — CHANGES_REQUESTED** [review]
See my comment around handling the `childSpan` and `histTimerEnd` before your return.

File: test/unit/handlers/parties/{Type}/{ID}/{SubId}.test.js (hunk 1 above)
Comment (bushjames) [reply test/unit/handlers/parties/{Type}/{ID}/{SubId}.test.js:74]: darn it!

File: src/domain/parties/parties.js (hunk 2 above)
Comment (bushjames) [reply src/domain/parties/parties.js:90]: thanks. will fix.

**Review by mdebarros — APPROVED** [review]
+1

Note of the Moderate Security issues on Shim's dependency have been ignored for a week.

Unfortunately, no fixes at the moment, see my audit-resolve fix --> https://github.com/mojaloop/account-lookup-service/pull/417/commits/a3057231de8647661bb3d5e16c242b032d2ddc75.

@elnyry-sam-k @lewisdaly I will be creating a story for this. The Shim issue is ongoing, and we will need to investigate how can we resolve this going forward. Hopefully, it is as simple as updating the dependency on the @mojaloop/central-services-shared. Luckily this is only impacting the self-hosted documentation end-point which should not be exposed publically --> https://github.com/mojaloop/project/issues/2354

**Comment by lewisdaly** [comment]
Thanks @bushjames , @mdebarros

**Pull request merged**